### ISSUE TITLE
Skip raising warning 'No authentication found' for Application Default Credentials

### ISSUE THREAD

**Issue by vmotso** [google-github-actions/setup-gcloud#646]
### TL;DR

We are using self-hosted runners with [Application Default Credentials](https://github.com/google-github-actions/setup-gcloud#application-default-credentials)
Everything works great, but we ran this action many times in parallel and got a lot of warnings in annotations:

![warnings](https://github.com/google-github-actions/setup-gcloud/assets/20564930/1b7fdfcd-96c0-4aac-a330-296dc7ae6e9f)

and we miss useful information that we print in the annotations.

### Expected behavior

We expect no warning to be printed for Application Default Credentials.

### Action YAML

```yaml
- name: Set up gcloud
  uses: google-github-actions/setup-gcloud@v1
  with:
    install_components: beta
```

### Additional information

Maybe something like this:
```yaml
    - name: Set up gcloud
      uses: google-github-actions/setup-gcloud@v1
      with:
        install_components: beta
        auth: skip
```
would be nice to have.

**Comment by sethvargo** [comment]
Duplicate of https://github.com/google-github-actions/setup-gcloud/issues/594.
